src/Security/Authentication/Core/src/RemoteAuthenticationHandler.cs

```diff
@@ -41,19 +48,47 @@ protected override Task<object> CreateEventsAsync()
         public virtual Task<bool> ShouldHandleRequestAsync()
             => Task.FromResult(Options.CallbackPath == Request.Path);
 
-        public virtual async Task<bool> HandleRequestAsync()
+        public virtual Task<bool> HandleRequestAsync()
```

Comment:
Not sure why you optimized this path when the goal of this PR is to avoid calling it at all.

Reply:
I optimized it for when users are opting out from using Endpoint.